### PR TITLE
Support for nested tokenPropertyName (modified)

### PR DESCRIPTION
**Pull request by eluciano11**
This pull request contains the commits made by @thhermansen on #124 . It does not include the upgrade to ember and to ember-cli. Upgrade to ember and ember-cli 1.13.15 were made on #136.

### CODE REVIEW

File: addon/authenticators/jwt.js [hunk 1]
```diff
@@ -158,24 +158,13 @@ export default TokenAuthenticator.extend({
                                  otherwise
   */
   authenticate(credentials, headers) {
-
     return new Ember.RSVP.Promise((resolve, reject) => {
-      const data = this.getAuthenticateData(credentials);
+      let data = this.getAuthenticateData(credentials);
```
Comment (jpadilla) [inline addon/authenticators/jwt.js]: Any reason why this can't be `const`?

File: addon/authenticators/jwt.js [hunk 2]
```diff
@@ -226,26 +215,14 @@ export default TokenAuthenticator.extend({
     @private
   */
   refreshAccessToken(token, headers) {
-    let data = {
-      [this.tokenPropertyName]: token
-    };
+    let data = this.makeRefreshData(token);
```
Comment (jpadilla) [inline addon/authenticators/jwt.js]: Any reason why this can't be `const`?

**Comment by thhermansen** [comment]
I think I fixed the failing test and changed `let` to `const`. Please see [this diff](https://github.com/eluciano11/ember-simple-auth-token/compare/master...thhermansen:eluciano11-master) between @eluciano11 master against my latest changes.

As far as I understood it, Qunit tear down and moved to the next test a little too fast and didn't wait for the test to complete it's assertions. I fixed this by doing an `assert.async();` and calling the async callack when the test was done.

Feel free to merge my changes is to @eluciano11 master branch so the changes will be included in to this pull request (as I don't think I can do that myself).

**Comment by eluciano11** [comment]
@homu r+

**Comment by eluciano11** [comment]
@thhermansen thank you for your work! Soon we'll release a new version of the add-on with your changes.

**Pull request merged**